service/controller/app/resource/validation/create.go

```diff
@@ -21,10 +21,22 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 
 	_, err = r.appValidator.ValidateApp(ctx, cr)
-	if validation.IsValidationError(err) {
+
+	if validation.IsAppConfigMapNotFound(err) || validation.IsKubeConfigNotFound(err) {
```

Comment:
Can we maybe also check what the current status is? E.g. if there is no status -> set pending, if there was a status (e.g. deployed) then it should be error? Does that make sense?

Reply:
@MarcelMue Yes that makes total sense to check if it's already installed.

This didn't work out in testing. Currently the status is empty instead of resource-not-found. 

I'm going to move this back to draft until I've tested some more. Sorry for the noise.